### ISSUE TITLE
Using only Phone OTP flow to login a user

### ISSUE THREAD

**Issue by LohithBlaze** [FusionAuth/fusionauth-issues#615]
### Problem
Since fusionauth provides sms otp as the second factor, is there a way for users to login to fusionauth by using only the phone OTP as the first factor? if not, this feature will be helpful for users to not provide any passwords.

### Solution
Provide users to login to fusionauth using only the phoneOTP flow as the first factor.

### Alternatives/workarounds
Right now, we need to have another auth service to achieve this feature.

### How to vote
thumbs up

**Comment by robotdan** [comment]
I thought we had an open issue for this, but I could find it.

I think what you're after is passwordless via SMS. Is that correct?https://github.com/FusionAuth/fusionauth-issues/issues/38

If I can find the issues I can link them here. But what we have in plan is to build sms templates so we can perform localization. And then we'll likely build an SMS transport to be able to integrate with any provider , and them we just can let you select email or sms for the passwordless login config .

**Comment by robotdan** [comment]
You could also do this today using the HYPR integration.

https://fusionauth.io/docs/v1/tech/identity-providers/hypr
https://www.hypr.com/guide/
https://www.hypr.com/fusionauth-passwordless-mfa/

**Comment by LohithBlaze** [comment]
yes, looking for passwordless via SMS. We dont want to use HYPR since we are forcing users to install their app as well.

**Comment by mooreds** [comment]
If you want to run an external service, you could also use the [passwordless API](https://fusionauth.io/docs/v1/tech/apis/passwordless):

* Call `/api/passwordless/start`
* Send the code via whatever SMS provider you want
* Redirect the user to a page where they can enter the code
* When they do, post to `/api/passwordless/login`
* Then send the corresponding token back to the client.

**Comment by jdtommy** [comment]
I would do what mooreds has said but the code is to long for a reoccurring login situation by a OTP/passwordless option. If I code supply the code for passwordless and let it be a more manageable value for typing in from a SMS that would be great

**Comment by mooreds** [comment]
You can control the length and type of the code. See https://fusionauth.io/docs/v1/tech/core-concepts/tenants#external-identifier-generation-form-fields

And here's an image of me setting the passwordless login value to be 6 digits.

<img width="765" alt="Screenshot 2023-07-06 at 8 04 33 PM" src="https://github.com/FusionAuth/fusionauth-issues/assets/91825/71451a06-82a3-4114-aaea-643fb9f1e457">

**Comment by mooreds** [comment]
@jdtommy

**Comment by robotdan** [comment]
This should be possible as part of the work being delivered under https://github.com/FusionAuth/fusionauth-issues/issues/1.

**Comment by robotdan** [comment]
Assuming this can be closed now? In version `1.59.0` we added phone as a first class identity. So you can now use phone with the Passwordless Login API. 

- https://github.com/FusionAuth/fusionauth-issues/issues/3145

Closing - please re-open if this use case is not correctly resolved via the linked issue in `1.59.0`.
